### PR TITLE
Add new membership procedure doc

### PR DESCRIPTION
**Pull request by cblecker**
/cc @kubernetes/owners

### CODE REVIEW

File: github-management/new-membership-procedure.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+All members of the Kubernetes GitHub org are required to do the following:
+
+- Have 2FA enabled on their GitHub account
+
+  This is enforced by GitHub. The contributor will not be able to accept the org
+  invitation without it, and they will be removed from the org automatically in
+  the event they disable it.
+
+- Join the k-dev mailing list
```
Comment (grodrigues3) [inline github-management/new-membership-procedure.md]: Link to the mailing list?

Comment (cblecker) [reply github-management/new-membership-procedure.md]: Added

Comment (cblecker) [reply github-management/new-membership-procedure.md]: Added

**Comment by spiffxp** [comment]
/approve
/hold
for @cblecker to /hold cancel

**Comment by cblecker** [comment]
Thanks all for reviews :)

/hold cancel

**Pull request merged**